portable/common/ff_ramdisk.c

```diff
@@ -419,10 +419,10 @@ BaseType_t FF_RAMDiskShowPartition( FF_Disk_t * pxDisk )
         FF_PRINTF( "Partition Nr   %8u\n", pxDisk->xStatus.bPartitionNumber );
         FF_PRINTF( "Type           %8u (%s)\n", pxIOManager->xPartition.ucType, pcTypeName );
         FF_PRINTF( "VolLabel       '%8s' \n", pxIOManager->xPartition.pcVolumeLabel );
-        FF_PRINTF( "TotalSectors   %8lu\n", pxIOManager->xPartition.ulTotalSectors );
-        FF_PRINTF( "SecsPerCluster %8lu\n", pxIOManager->xPartition.ulSectorsPerCluster );
-        FF_PRINTF( "Size           %8lu KB\n", ulTotalSizeKB );
-        FF_PRINTF( "FreeSize       %8lu KB ( %d perc free )\n", ulFreeSizeKB, iPercentageFree );
+        FF_PRINTF( "TotalSectors   %8lu\n", (unsigned long) pxIOManager->xPartition.ulTotalSectors );
+        FF_PRINTF( "SecsPerCluster %8lu\n", (unsigned long) pxIOManager->xPartition.ulSectorsPerCluster );
+        FF_PRINTF( "Size           %8lu KB\n", (unsigned long) ulTotalSizeKB );
+        FF_PRINTF( "FreeSize       %8lu KB ( %d perc free )\n", (unsigned long) ulFreeSizeKB, iPercentageFree );
```

Comment:
In stead of casting to `unsigned long`, the +FAT library normally uses `unsigned` in combination with `%u`:
~~~diff
-    FF_PRINTF( "TotalSectors   %8lu\n", (unsigned long) pxIOManager->xPartition.ulTotalSectors );
+    FF_PRINTF( "TotalSectors   %8u\n", (unsigned) pxIOManager->xPartition.ulTotalSectors );
~~~
It also removes the warning and it works well on 64-bit platforms as well.
Thank you

Reply:
Until now, I didn't comment on this PR ( except about `FF_PRINTF()` ) because I am totally inexperienced with cmake ( I only use GNU make ).

I checked out your branch and typed "cmake", which doesn't work obviously.
Would you mind to give a short description of how I can compile the project for dummies?

Also, if you open this PR, you will see that "This branch is out-of-date with the base branch", with a button "Update branch". Could you update it with the master branch?

I will talk to the team and see who can further review this PR. Sorry for the long delay.

Reply:
@htibosch -

- Short description on how to compile the project:

I've added a build check in ci/cd environment similar to the `FreeRTOS-Plus-TCP` repo (the .github directory is a copy and modify from that).

I've also updated the main README.md as to how to integrate with CMake in a higher level project.

Reply:
> As for the FF_PRINTF - all but one of the ff_ramdisk.c files are using %8lu rather than what you requested

That is possibly true, but we would like to get rid of all `%ld` and `%lu` formats, both in FreeRTOS+TCP and in FreeRTOS+TCP.

Reply:
@htibosch - My goal was to add the necessary code to have a linux port build similar to other FreeRTOS git repos (Kernel and Plus-TCP) and follow the same coding style and usage of the existing ports inside this `Plus-FAT` repo.  Changing some code to reflect an arbitrary request to get rid of a particular format in this and other repos is out of scope (from my perspective) of this particular PR.

Since this is an unrelated issue to any of the CMake build support (a different feature/clean-up task), I'll add it to a follow on PR associated with fixing compile warnings of this repo instead (and I'll change all of the various portable code instead of just this new Linux port).

See similar ones for:
- `FreeRTOS-Kernel`: https://github.com/phelter/FreeRTOS-Kernel/pull/1 
  - waiting on: https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/571
- `FreeRTOS-Plus-TCP`: https://github.com/phelter/FreeRTOS-Plus-TCP/compare/feature/cmake-support...feature/fixing-clang-gnu-compiler-warnings 
  - waiting on: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/557
- `FreeRTOS-Plus-FAT`: - will do once have this PR completed.

I intend to update the above ones and provide them, but have been blocked in continuing with this effort due to the outstanding pull requests.